.github/workflows/build_wheels.yml

```diff
@@ -0,0 +1,249 @@
+
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+          architecture: ${{ matrix.architecture }}
+
+      - name: Install Miniconda
+        uses: conda-incubator/setup-miniconda@v3
+        with:
+          auto-update-conda: true
+          python-version: "3.10"
+          architecture: ${{ matrix.architecture }}
+
+      - name: Install SLEEF and other dependencies
+        shell: bash -l {0}
+        run: |
+          conda config --add channels conda-forge
+          conda config --set channel_priority strict
+          conda install -y sleef numpy
+
+      - name: Setup build environment
+        shell: pwsh
+        run: |
+          $env:INCLUDE += ";$env:CONDA_PREFIX\Library\include"
+          $env:LIB += ";$env:CONDA_PREFIX\Library\lib"
+          $env:PATH = "$env:CONDA_PREFIX\Library\bin;$env:PATH"
+          echo "INCLUDE=$env:INCLUDE" >> $env:GITHUB_ENV
+          echo "LIB=$env:LIB" >> $env:GITHUB_ENV
+          echo "PATH=$env:PATH" >> $env:GITHUB_ENV
+
+      - name: Install build dependencies
+        shell: bash -l {0}
+        run: |
+          pip install -U pip
+          pip install cibuildwheel==2.20.0 ninja meson meson-python numpy delvewheel pytest
+
+      - name: Build wheels
+        env:
+          CIBW_BUILD: "cp310-* cp311-* cp312-*"
+          CIBW_SKIP: "pp* cp36-* cp37-* cp38-* cp39-* cp313-*"
+          CIBW_ARCHS_WINDOWS: ${{ matrix.architecture == 'x86' && 'x86' || 'AMD64' }}
+          CIBW_BUILD_VERBOSITY: "3"
+          DISTUTILS_USE_SDK: "1"
+          MSSdk: "1"
+          CIBW_BEFORE_BUILD: |
+            pip install meson meson-python ninja numpy
+          CIBW_REPAIR_WHEEL_COMMAND: "delvewheel repair -w {dest_dir} {wheel}"
+          CIBW_TEST_COMMAND: |
+            pip install {package}[test]
+            python -m pytest -v {project}/test
+          CIBW_TEST_EXTRAS: test
+          CIBW_TEST_FAIL_FAST: 1
+        shell: pwsh
+        run: |
+          python -m cibuildwheel --output-dir wheelhouse
+          if (-not (Test-Path wheelhouse/*.whl)) { throw "Wheel was not created" }
+        working-directory: ./quaddtype
+
+      - uses: actions/upload-artifact@v4
+        with:
+          path: ./quaddtype/wheelhouse/*.whl
+          name: wheels-windows-${{ matrix.architecture }}
+
+  publish_to_testpypi:
+    name: Publish to TestPyPI
+    needs: [build_wheels_linux, build_wheels_macos, build_wheels_windows]
+    runs-on: ubuntu-latest
+    if: startsWith(github.ref, 'refs/tags/quaddtype-v')
+    steps:
+      - name: Download all workflow run artifacts
+        uses: actions/download-artifact@v4
+        with:
+          path: dist
+      - name: Publish to TestPyPI
+        uses: pypa/gh-action-pypi-publish@v1.9.0
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_API_TOKEN }}
+          repository-url: https://test.pypi.org/legacy/
+          packages-dir: dist/*
```

Comment:
@mattip when you have time, can you coordinate with Swayam to add the configuration necessary to upload to pypi at https://pypi.org/project/numpy_quaddtype/? I think [these are the relevant docs](https://docs.pypi.org/trusted-publishers/adding-a-publisher/).

@SwayamInSync you should also probably manually upload a numpy_quaddtype release to pypi with a tarball containingnthe current state of the main branch. Mostly just to claim the name, but I think the project may have to exist already on pypi before you can set up trusted publishing.